packages/core/src/browser/widgets/dropdown-component.tsx

```diff
@@ -0,0 +1,153 @@
+/********************************************************************************
+ * Copyright (C) 2023 Hundsun and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import '../../../src/browser/style/dropdown-component.css';
+
+export interface DropdownOption {
+    label: string | React.ReactElement;
+    value: string;
+    group?: string;
+    onClick?: (e?: React.MouseEvent<HTMLElement>) => void
+}
+
+export interface DropdownComponentProps {
+    options: DropdownOption[];
+    children: React.ReactElement;
+    onBlur?: () => void,
+    onFocus?: () => void
+}
+
+export const DROPDOWN_COMPONENT_CONTAINER = 'dropdown-component-container';
+
+export const DropdownComponent: React.FC<DropdownComponentProps> = props => {
```

Comment:
This dropdown doesn't align with the CSS of our other menus. Would there be a way to automatically apply the styling of the phosphor menus to this dropdown or reuse those html elements instead?